### PR TITLE
Add OAuth token-based authentication to guide

### PR DESCRIPTION
**Pull request by XuanWang-Amos**
Add guide for OAuth token-based authentication

### CODE REVIEW

File: content/en/docs/guides/auth.md
Comment (matthewstevenson88) [inline content/en/docs/guides/auth.md:126]: IIUC, this library is primarily for sending an existing OAuth tokens around, refreshing them, etc and it doesn't deal with generating/obtaining OAuth tokens (except for the subpackage https://pkg.go.dev/golang.org/x/oauth2/google, which is Google-specific).

Similarly the `google-auth` link below is Google-specific.

Since this top section is generic and not Google-specific, should we just assume that the user already has an existing OAuth token? WDYT?

File: content/en/docs/guides/auth.md
Comment (matthewstevenson88) [inline content/en/docs/guides/auth.md:129]: If you want, you can replace "metadata" with the "HTTP Authorization header".

File: content/en/docs/guides/auth.md
Comment (matthewstevenson88) [inline content/en/docs/guides/auth.md:128]: nit: I think the OAuth token is always part of the per-call credentials, and then you might attach the per-call credentials to some channel credentials.

File: content/en/docs/guides/auth.md
Comment (matthewstevenson88) [inline content/en/docs/guides/auth.md:238]: This link points to Java, not Go.

**Review by matthewstevenson88 — COMMENTED** [review]
Thanks so much for adding this documentation! Left a few minor comments. :)

File: content/en/docs/guides/auth.md
Comment (XuanWang-Amos) [reply content/en/docs/guides/auth.md:126]: I see, I think that make sense, let me rephrase this part.

File: content/en/docs/guides/auth.md
Comment (XuanWang-Amos) [reply content/en/docs/guides/auth.md:128]: Got it, I'll add this.

File: content/en/docs/guides/auth.md
Comment (XuanWang-Amos) [reply content/en/docs/guides/auth.md:129]: Make sense, I'll use header instead since it's more accurate.

File: content/en/docs/guides/auth.md
Comment (XuanWang-Amos) [reply content/en/docs/guides/auth.md:238]: Nice catch!

**Review by XuanWang-Amos — COMMENTED** [review]
Thanks for reviewing!

**Review by matthewstevenson88 — APPROVED** [review]
Looks great! Thanks again for making these changes.

Please also get an approval from one of the leads, since I don't believe I have write access for grpc.io. :)

File: content/en/docs/guides/auth.md
Comment (temawi) [inline content/en/docs/guides/auth.md:129]: "will be sent"

**Review by temawi — APPROVED** [review]
Just one typo to fix.

**Comment by temawi** [comment]
Thanks Xuan, I'll merge it.

**Comment by Paullevey** [comment]
No  how can my acount stay 0    its not.  Corectt

**Comment by Paullevey** [comment]
All assets is belangs to to me NOT  x  (VERONICA BESTER) SHE KEEP HANKEDING ME

**Pull request merged**
